server/app/views/questiontypes/ApplicantCompositeQuestionRenderer.java

```diff
@@ -0,0 +1,57 @@
+package views.questiontypes;
+
+import static j2html.TagCreator.fieldset;
+import static j2html.TagCreator.legend;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import j2html.tags.specialized.DivTag;
+import org.apache.commons.lang3.StringUtils;
+import services.Path;
+import services.applicant.ValidationErrorMessage;
+import services.applicant.question.ApplicantQuestion;
+import views.style.ApplicantStyles;
+import views.style.ReferenceClasses;
+
+/*
+ * Superclass for questions that have multiple input fields.
+ *
+ * Questions with multiple input fields should use a fieldset and legend for a11y. Question level
+ * errors and descriptions are attached at the fieldset level, instead of to individual inputs.
+ */
+abstract class ApplicantCompositeQuestionRenderer extends ApplicantQuestionRendererImpl {
+
+  ApplicantCompositeQuestionRenderer(ApplicantQuestion question) {
+    super(question);
+  }
+
+  protected abstract DivTag renderInputTag(
```

Comment:
question (non-blocking): should this be pluralized?

Reply:
Yes, that SG! Done.